js/ui/codeView.js

```diff
@@ -95,6 +96,25 @@ const _ensureHackDataFile = (function () {
         if (initialized)
             return keyfile;
 
+        // Only create file monitors the first time
+        if (monitors.length === 0) {
```

Comment:
Rather than checking this, you can just move this block up to just before `return function() {`.

Reply:
We can't, if we do that, the monitors variable doesn't have any reference and all monitors are destroyed. I tried and it takes me a while to understand why it was not working.

If we want to do that outside we need to move the monitors variable to the global scope

Reply:
This seems really fragile: is it actually guaranteed that variables in the outer function scope which are not referenced from the inner function stay alive?

Reply:
But, OK.